DataFormats/Luminosity/interface/LumiInfo.h

```diff
@@ -15,70 +15,170 @@
  *         Zhen Xie
  *         Paul Lujan
  * \version   1st Version June 7 2007, merged September 10 2014
+ * \version    October 2017 by Chris Palmer and Sam Higginbotham for PCC projects
+ * 
  *
- ************************************************************/
+ */
  
 #include <vector>
 #include <iosfwd>
 #include <string>
 #include "DataFormats/Luminosity/interface/LumiConstants.h"
 
 class LumiInfo {
-public:
-  /// default constructor
-  LumiInfo():
-  deadtimeFraction_(0)
-  { 
-    instLumiByBX_.assign(LumiConstants::numBX, 0.0);
-  } 
+  public:
+    /** 
+     * default constructor
+     */
+    LumiInfo():
+    deadtimeFraction_(0)
+    { 
+        instLumiByBX_.assign(LumiConstants::numBX, 0.0);
+        instLumiStatErrByBX_.assign(LumiConstants::numBX, 0.0);
+        totalInstLuminosity_=0;
+        totalInstLumiStatErr_=0;
+    } 
   
-  /// constructor with fill
- LumiInfo(float deadtimeFraction,
-	  const std::vector<float>& instLumiByBX):
-  deadtimeFraction_(deadtimeFraction)
-  {
-    instLumiByBX_.assign(instLumiByBX.begin(), instLumiByBX.end());
-  }
 
-  /// destructor
-  ~LumiInfo(){}
+    /** 
+     * constructor with fill; if total algo is the same as summing
+     */
+    LumiInfo(float deadtimeFraction, const std::vector<float>& instLumiByBX):
+    deadtimeFraction_(deadtimeFraction)
+    {
+        instLumiByBX_.assign(instLumiByBX.begin(), instLumiByBX.end());
+        instLumiStatErrByBX_.assign(LumiConstants::numBX, 0.0);
+        setTotalInstToBXSum() ;
+        totalInstLumiStatErr_=0;
+    }
 
-  // Instantaneous luminosity (in Hz/ub)
-  float instLuminosity() const;
+    /** 
+     * constructor with fill; if total algo DIFFERS from summing
+     */
+    LumiInfo(float deadtimeFraction, const std::vector<float>& instLumiByBX, float totalInstLumi):
+    deadtimeFraction_(deadtimeFraction)
+    {
+        instLumiByBX_.assign(instLumiByBX.begin(), instLumiByBX.end());
+        instLumiStatErrByBX_.assign(LumiConstants::numBX, 0.0);
+        totalInstLuminosity_=totalInstLumi;
+        totalInstLumiStatErr_=0;
+    }
 
-  // Integrated (delivered) luminosity (in ub^-1)
-  float integLuminosity() const;
+    /** 
+     * constructor with fill; if total algo DIFFERS from summing and adding including stats
+     */
+    LumiInfo(float deadtimeFraction, const std::vector<float>& instLumiByBX, float totalInstLumi,
+    const std::vector<float>& instLumiErrByBX, float totalInstLumiErr):
+    deadtimeFraction_(deadtimeFraction)
+    {
+        instLumiByBX_.assign(instLumiByBX.begin(), instLumiByBX.end());
```

Comment:
replacing this with instLumiByBX_(instLumiByBX) two lines above might be more clear.